### ISSUE TITLE
Properties of IDataContext are not evaluated for ExpressionMethod

### ISSUE THREAD

**Issue by Feroks** [linq2db/linq2db#2226]
I have defined a column that uses `ExpressionMethod`. It relies on additional input parameter, in my case it is `UserId`. 
In order to simplify things i decided to add it as a property on `DataContext` class and then use it inside the `ExpressionMethod` implementation (see the example below). It fails, because linq2db tries to create SQL code for it instead of using the value directly.

Is this by design? How can i achieve my goal?

```
Exception message:
Convert(dctx).CurrentUserId' cannot be converted to SQL.
Stack trace:
LinqToDB.Linq.LinqException: 'Convert(dctx).CurrentUserId' cannot be converted to SQL.
  at LinqToDB.Linq.Builder.ExpressionBuilder.ConvertToSql(IBuildContext context, Expression expression, Boolean unwrap)
  at LinqToDB.Linq.Builder.ExpressionBuilder.ConvertCompare(IBuildContext context, ExpressionType nodeType, Expression left, Expression right)
  at LinqToDB.Linq.Builder.ExpressionBuilder.ConvertPredicate(IBuildContext context, Expression expression)
  at LinqToDB.Linq.Builder.ExpressionBuilder.BuildSearchCondition(IBuildContext context, Expression expression, List`1 conditions, Boolean isNotExpression)
  at LinqToDB.Linq.Builder.ExpressionBuilder.BuildWhere(IBuildContext parent, IBuildContext sequence, LambdaExpression condition, Boolean checkForSubQuery, Boolean enforceHaving)
  at LinqToDB.Linq.Builder.WhereBuilder.BuildMethodCall(ExpressionBuilder builder, MethodCallExpression methodCall, BuildInfo buildInfo)
  at LinqToDB.Linq.Builder.MethodCallBuilder.BuildSequence(ExpressionBuilder builder, BuildInfo buildInfo)
  at LinqToDB.Linq.Builder.ExpressionBuilder.BuildSequence(BuildInfo buildInfo)
  at LinqToDB.Linq.Builder.CountBuilder.BuildMethodCall(ExpressionBuilder builder, MethodCallExpression methodCall, BuildInfo buildInfo)
  at LinqToDB.Linq.Builder.MethodCallBuilder.BuildSequence(ExpressionBuilder builder, BuildInfo buildInfo)
  at LinqToDB.Linq.Builder.ExpressionBuilder.BuildSequence(BuildInfo buildInfo)
  at LinqToDB.Linq.Builder.ExpressionBuilder.Build[T]()
  at LinqToDB.Linq.Query`1.CreateQuery(IDataContext dataContext, Expression expr)
```

### Steps to reproduce
Include a complete code listing (or project/solution) that we can run to reproduce the issue.

Partial code listings, or multiple fragments of code, will slow down our response or cause us to push the issue back to you to provide code to reproduce the issue.

```c#
[TestFixture]
public class Issue : TestBase
{
	public class Entity
	{
		[Column]
		public int UserId { get; set; }

		[ExpressionMethod(nameof(BelongsToCurrentUserExpr))]
		public bool BelongsToCurrentUser { get; set; }
		
		public static Expression<Func<Entity, CustomDataContext, bool>> BelongsToCurrentUserExpr()
		{
			return (e, db) => e.UserId == db.CurrentUserId;
		}
	}
	
	[Test]
	public void Test([IncludeDataSources(TestProvName.AllSQLite)] string context)
	{
		using (var db = new CustomDataContext(context))
		using (db.CreateLocalTable(new[]
		{
			new Entity {UserId = 1},
			new Entity {UserId = 2},
			new Entity {UserId = 2}
		}))
		{
			db.CurrentUserId = 1;
			var count = db
				.GetTable<Entity>()
				.Count(x => x.BelongsToCurrentUser);
			
			Assert.AreEqual(1, count);

			db.CurrentUserId = 2;
			count = db
				.GetTable<Entity>()
				.Count(x => x.BelongsToCurrentUser);
			
			Assert.AreEqual(2, count);
		}
	}
	
	public class CustomDataContext : DataContext
	{
		public CustomDataContext(string? configurationString) : base(configurationString)
		{
		}

		public int CurrentUserId { get; set; }
	}
}
```

### Environment details
linq2db version: 3.0.0-preview2
Database Server: Sql Server
Database Provider: SqlServer
Operating system: Windows 10
.NET Framework: 4.6.2

**Comment by sdanyliv** [comment]
Currently it is not supported to use DataContext properties, will implement that in next release.

**Comment by sdanyliv** [comment]
Just for clarification. Why you have created descendant from `DataContext`? It is preferred to make descendant from `DataConnection`

**Comment by Feroks** [comment]
It is by accident. I am using `DataConnection` in my real code.

**Comment by MaceWindu** [comment]
Fixed by #2228
